### ISSUE TITLE
VGroups must support item assignment

### ISSUE THREAD

**Issue by Aathish04** [ManimCommunity/manim#1530]
## Enhancement proposal

Currently, VGroups are like Tuples and Strings in that they don't support item assignment. There's no good reason for them to remain this way.

Adding item assignments would let the user do things like:

```py
        nodes = VGroup([Dot(),Dot(),...,Dot()])
        nodedot3d = Dot3D(nodes[5].get_center(),color=RED)
        self.play(
                ReplacementTransform(nodes[5],nodedot3d)]
            )
        nodes[5] = nodedot3d
```
as opposed to what they would have to do currently:

```py
        nodes = VGroup([Dot(),Dot(),...,Dot()])
        nodedot3d = Dot3D(nodes[5].get_center(),color=RED)
        self.play(
                ReplacementTransform(nodes[5],nodedot3d)]
            )
        nodes = VGroup(*nodes[:5],nodedot3d,*nodes[6:])
```

**Comment by AntonBallmaier** [comment]
Item assignment would need to include a check that only VMobjects are passed. Also, wouldn't this apply to `Group` as well?

**Comment by ryanmccauley211** [comment]
I would like to start contributing, so I can take this as my first task.

**Comment by Aathish04** [comment]
Closed by #1559.
